### PR TITLE
docker_sd: Support host network mode

### PR DESCRIPTION
**Pull request by LeviHarrison**
This PR adds a new configuration option, `host_network_host` (I'm kind of still unsure about this name), that is used as the host when the container is set to host network mode.

Fixes: #9116

### CODE REVIEW

File: discovery/moby/docker.go [hunk 1]
```diff
@@ -245,7 +249,15 @@ func (d *DockerDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group, er
 					labels[model.LabelName(k)] = model.LabelValue(v)
 				}
 
-				addr := net.JoinHostPort(n.IPAddress, strconv.FormatUint(uint64(d.port), 10))
+				// Containers in host networking mode don't have ports,
+				// so they only end up here, not in the previous loop.
+				var addr string
+				if c.HostConfig.NetworkMode != "host" {
+					addr = net.JoinHostPort(n.IPAddress, strconv.FormatUint(uint64(d.port), 10))
```
Comment (roidelapluie) [inline discovery/moby/docker.go:256]: can we just look if n.IPAddress is empty?

Comment (LeviHarrison) [reply discovery/moby/docker.go:256]: I guess we could, but this way is more verbose and logical. There could foreseeably be a bug or misconfiguration, although unlikely, that causes the IP address to be empty when host networking mode _is_ on, and in that case, the SD should fail in the current manner it does for host networking.

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -639,6 +639,9 @@ tls_config:
 # tasks and services that don't have published ports.
 [ port: <int> | default = 80 ]
 
+# The host to use if the container is in host networking mode.
+[ host_network_host: <string> | default = "localhost" ]
```
Comment (roidelapluie) [inline docs/configuration/configuration.md:643]:
```suggestion
[ host_networking_host: <string> | default = "localhost" ]
```

I think docker refers to this as host networking everywhere.

**Comment by roidelapluie** [comment]
Thanks!

We have 5 weeks to find a less confusing option than `host_networking_host`

**Comment by LeviHarrison** [comment]
Yeah, it's pretty bad. Unfortunately we are configuring what can best be called a "host" for something that starts with "host".

Open to ideas, I can't think of anything else.

**Pull request merged**